assets/js/types/type-defs/api-response.ts

```diff
@@ -0,0 +1,37 @@
+export interface ApiResponse< T > {
+	body: Record< string, unknown >;
+	headers: Headers;
+	status: number;
+	ok: boolean;
+	json: () => Promise< T >;
+}
+
+export function assertBatchResponseIsValid(
+	response: unknown
+): asserts response is {
+	responses: ApiResponse< unknown >[];
+	headers: Headers;
+} {
+	if (
+		typeof response === 'object' &&
+		response !== null &&
+		response.hasOwnProperty( 'responses' )
+	) {
+		return;
+	}
+	throw new Error( 'Response not valid' );
+}
+
+export function assertResponseIsValid< T >(
```

Comment:
This fails when submitting the checkout form, causing the order to be placed correctly on the server, but the client shows an error.

Reply:
A checkout response looks like this
<img width="712" alt="image" src="https://user-images.githubusercontent.com/5656702/204620947-e4f1f2e8-a781-4ca6-b6aa-bef8bcc1a3f7.png">

Reply:
Fixed